app/questionnaire/navigator.py

```
@@ -133,8 +133,11 @@ def build_path(self, blocks, group_id, group_instance, block_id, path):
         path.append(this_block)
 
         # Return the index of the block id to be visited
-        block_id_index = next(index for (index, b) in enumerate(blocks) if b["block"]["id"] == block_id and
-                              b["group_id"] == group_id and b['group_instance'] == group_instance)
+        block_id_index = next((index for (index, b) in enumerate(blocks) if b["block"]["id"] == block_id and
+                              b["group_id"] == group_id and b['group_instance'] == group_instance), None)
+
+        if block_id_index is None:
```

Wouldn't it be better to throw an error here and catch/log at the point the path was built? Otherwise the recursive call will still carry on infinitely, just logging.

---

Is there any point catching it? i know thats not usually very good, but what are we gonna do to recover in this event?

---

I don't know what we'd do to recover, but it would save a horrible recursive call as soon as we know when we can't build a path any further.